lib/utils/cli.go

```diff
@@ -114,9 +114,44 @@ func GetIterations() int {
 	return iter
 }
 
-// UserMessageFromError returns user friendly error message from error
+// UserMessageFromError returns user-friendly error message from error.
+// The error message will be formatted for output depending on the debug
+// flag
 func UserMessageFromError(err error) string {
-	// untrusted cert?
+	if err == nil {
+		return ""
+	}
+	if certErr := formatCertError(err); certErr != "" {
+		return certErr
+	}
+	if log.GetLevel() == log.DebugLevel {
+		return trace.DebugReport(err)
+	}
+	return FormatError(err)
+}
+
+// FormatError returns user friendly error message from error.
+// The error message is escaped if necessary
+func FormatError(err error) string {
```

Comment:
Seems like the only difference b/w this new function and `UserMessageFromError` above is that the one above also returns a stack trace in debug mode? Any reason that above function can't just be used everywhere (with your new `\n` logic) instead of having two very similar ones, or am I missing anything?

Reply:
> Seems like the only difference b/w this new function and `UserMessageFromError` above is that the one above also returns a stack trace in debug mode? Any reason that above function can't just be used everywhere (with your new `\n` logic) instead of having two very similar ones, or am I missing anything?

Yes, with `FormatError` I tried to use the same functionality in the old `UserMessageFromError` while deviating from the implicit debug prints which are problematic as shown in [this](https://github.com/gravitational/teleport/issues/3205) issue. The latter is still used universally to output errors so I did not want to alter its functionality at risk of breaking anything.